classes/07class/kms/README.md

```diff
@@ -0,0 +1,106 @@
+
+For each type of key a list of types and algorithms are supported and can be found in the [KMS FAQ page](https://aws.amazon.com/kms/faqs/).
+
+## How to use KMS?
+KMS is easy to use and in most cases you can encrypt/decrypt your data without having to interact with the KMS service directly. When creating resources(through console, API or terraform) you can specify if encryption is enabled or disabled. In this situation, you're already making use of KMS indirectly.
+
+For example, when creating an EBS volume in terraform:
+```terraform
+resource "aws_ebs_volume" "example" {
+  availability_zone = var.az
+  size              = var.disk_size
+  encrypted         = true
+
+  tags = {
+    Name = var.name
+  }
+}
+```
+In the example above, the EBS volume being created will use a KMS key to be ecnrypted. Since we're not specyifying the key to be used through the `kms_key_id` argument, the resource will be created using the default KMS key for it's service(ebs in this case).
+
+Using the console, it's easier to see which key is being used in the case that no key is provided:
+![kms-key](assets/kms-key.png)
+
+Because it's not possible to have a lot of controls over the service default's key, it's recommended to have your own keys to increase the security being applied to the resources. 
+
+The terraform code below is using a custom key to encrypt the ebs volume.
+
+```terraform
+resource "aws_kms_key" "my_key" {
+  description = "My KMS Key"
+}
+
+resource "aws_ebs_volume" "example" {
+  availability_zone = var.az
+  size              = var.disk_size
+  encrypted         = true
+  kms_key_id        = aws_kms_key.my_key.arn
+  tags = {
+    Name = var.name
+  }
+}
+```
+
+With that approach, by using AWS IAM it's possible to create policies to restrict which users/roles can perform encrypt/decrypt action using that key.
+
+## Key's Management
+For the AWS managed key (service's defaults), there is nothing to do. Keys will be fully managed by AWS and will be rotated every three years. There is no way to change/rotate/expire/delete those keys.
+
+For Customer Managed Keys (created by users), there is minimum management, but you have more control over the key lifecycle. For example, its possible to expire and delete a customer manager key. The key rotation can be done by the use or can be configured to be rotated automatically by AWS, which will rotate the key every year. 
+
+The example below shows a key that will be automatically rotated by AWS:
+
+```terraform
+resource "aws_kms_key" "my_key" {
+  description         = "My KMS Key"
+  enable_key_rotation = true
+}
+```
+
+Because customer managed keys can be expired and deleted, is important to note that the key deletion action is destructive and potentially dangerous and must be done only when you have absolute sure that there is no data encrypted with that key.
```

Comment:
```suggestion
Because customer-managed keys can be expired and deleted, is important to note that the key deletion action is destructive and potentially dangerous and must be done only when you are absolutely sure that there is no data encrypted with that key.
```